File: test-api.py
#!/usr/bin/env python3
import requests
import json

base_url = "http://localhost:5000"

print("=" * 60)
print("Step 1: Authenticating...")
print("=" * 60)

# Login
login_response = requests.post(
    f"{base_url}/api/auth/login",
    json={"username": "admin", "password": "password"}
)
token = login_response.json().get("token")
print(f"✅ Token: {token[:30]}...\n")

print("=" * 60)
print("Step 2: Sending test pipeline data...")
print("=" * 60)

headers = {
    "Authorization": f"Bearer {token}",
    "Content-Type": "application/json"
}

# Send test data
test_payload = {
    "workflow": "CI/CD Pipeline",
    "status": "passed",
    "commit": "abc123",
    "branch": "main",
    "buildNumber": 1,
    "author": "test"
}

response = requests.post(
    f"{base_url}/api/admin/pipeline-status",
    json=test_payload,
    headers=headers
)
print(f"Response: {json.dumps(response.json(), indent=2)}\n")

print("=" * 60)
print("Step 3: Fetching pipeline stats...")
print("=" * 60)

stats_response = requests.get(
    f"{base_url}/api/admin/pipeline-status?limit=50",
    headers=headers
)
stats_data = stats_response.json()

print(f"Total Pipelines: {stats_data['stats']['totalPipelines']}")
print(f"Passed: {stats_data['stats']['passedPipelines']}")
print(f"Failed: {stats_data['stats']['failedPipelines']}")
print(f"Success Rate: {stats_data['stats']['successRate']}\n")

print("=" * 60)
print("✅ TEST COMPLETED!")
print("Now refresh your admin dashboard at http://localhost:3000")
print("=" * 60)

File: test-api.js
const http = require('http');

function makeRequest(method, path, data, token) {
  return new Promise((resolve, reject) => {
    const options = {
      hostname: 'localhost',
      port: 5000,
      path: path,
      method: method,
      headers: {
        'Content-Type': 'application/json',
      }
    };

    if (token) {
      options.headers['Authorization'] = `Bearer ${token}`;
    }

    const req = http.request(options, (res) => {
      let body = '';
      res.on('data', chunk => body += chunk);
      res.on('end', () => {
        try {
          resolve(JSON.parse(body));
        } catch (e) {
          resolve(body);
        }
      });
    });

    req.on('error', reject);
    if (data) req.write(JSON.stringify(data));
    req.end();
  });
}

async function test() {
  console.log('='.repeat(60));
  console.log('Step 1: Authenticating...');
  console.log('='.repeat(60));

  const loginResponse = await makeRequest('POST', '/api/auth/login', {
    username: 'admin',
    password: 'password'
  });

  const token = loginResponse.token;
  console.log(`✅ Token: ${token.substring(0, 30)}...\n`);

  console.log('='.repeat(60));
  console.log('Step 2: Sending test pipeline data...');
  console.log('='.repeat(60));

  const testResponse = await makeRequest('POST', '/api/admin/pipeline-status', {
    workflow: 'CI/CD Pipeline',
    status: 'passed',
    commit: 'abc123',
    branch: 'main',
    buildNumber: 1,
    author: 'test'
  }, token);

  console.log('Response:', JSON.stringify(testResponse, null, 2), '\n');

  console.log('='.repeat(60));
  console.log('Step 3: Fetching pipeline stats...');
  console.log('='.repeat(60));

  const statsResponse = await makeRequest('GET', '/api/admin/pipeline-status?limit=50', null, token);

  console.log(`Total Pipelines: ${statsResponse.stats.totalPipelines}`);
  console.log(`Passed: ${statsResponse.stats.passedPipelines}`);
  console.log(`Failed: ${statsResponse.stats.failedPipelines}`);
  console.log(`Success Rate: ${statsResponse.stats.successRate}\n`);

  console.log('='.repeat(60));
  console.log('✅ TEST COMPLETED!');
  console.log('Now refresh your admin dashboard at http://localhost:3000');
  console.log('='.repeat(60));
}

test().catch(console.error);
